library/Vanilla/Permissions.php

```diff
@@ -274,6 +293,21 @@ public function setBanned($isBanned) {
     }
 
     /**
+     * Get the flag to determine if all permission checks are being overridden.
+     *
+     * @param bool|null $overrideAll
+     * @return $this
+     */
+    public function setOverrideAll($overrideAll) {
+        if ($overrideAll !== null) {
+            $overrideAll = boolval($overrideAll);
```

Comment:
I view this as an easy-to-implement piece of defensive programming. It will ensure that, no matter what value is received by `setOverrideAll`, this class can always be certain its `overrideAll` property will be a boolean or null value. The class, and other parts of Vanilla that use its `getOverrideAll` method, can always use identical comparisons for reliable results.

I don't see a downside to making an open-source application more resilient with minimal effort.